src/Components/Resource/ListView.tsx

```diff
@@ -17,48 +12,34 @@ import { useTranslation } from "react-i18next";
 import { AdvancedFilterButton } from "../../CAREUI/interactive/FiltersSlideover";
 import CareIcon from "../../CAREUI/icons/CareIcon";
 import dayjs from "../../Utils/dayjs";
+import useQuery from "../../Utils/request/useQuery";
+import routes from "../../Redux/api";
 
 const Loading = lazy(() => import("../Common/Loading"));
 const PageTitle = lazy(() => import("../Common/PageTitle"));
 
 export default function ListView() {
-  const dispatch: any = useDispatch();
   const { qParams, Pagination, FilterBadges, advancedFilter, resultsPerPage } =
     useFilters({});
-  const [data, setData] = useState<any[]>([]);
-  const [totalCount, setTotalCount] = useState(0);
-  const [isLoading, setIsLoading] = useState(false);
   const { t } = useTranslation();
 
   const onBoardViewBtnClick = () =>
     navigate("/resource/board", { query: qParams });
   const appliedFilters = formatFilter(qParams);
 
   const refreshList = () => {
-    fetchData();
+    refetch();
   };
 
-  const fetchData = () => {
-    setIsLoading(true);
-    dispatch(
-      listResourceRequests(
-        formatFilter({
-          ...qParams,
-          offset: (qParams.page ? qParams.page - 1 : 0) * resultsPerPage,
-        }),
-        "resource-list-call"
-      )
-    ).then((res: any) => {
-      if (res && res.data) {
-        setData(res.data.results);
-        setTotalCount(res.data.count);
-      }
-      setIsLoading(false);
-    });
-  };
+  const { loading, data, refetch } = useQuery(routes.listResourceRequests, {
+    query: formatFilter({
+      ...qParams,
+      offset: (qParams.page ? qParams.page - 1 : 0) * resultsPerPage,
+    }),
+  });
 
   useEffect(() => {
-    fetchData();
+    refetch();
   }, [
```

Comment:
You don't need a `useQuery` to invoke `refetch` when `qParams` changes. `useQuery` itself will detect that automatically and do a `refetch` automatically.